### PR TITLE
Additional pre-submission updates

### PR DESCRIPTION
**Pull request by serbinsh**
Updated R_Scripts/functions.R to revise names of component selection options. Updated R_Scripts/expanded_spectra-trait_reseco_lma_plsr_example.R to allow for selection of custom output.  Added additional output plotting.

### CODE REVIEW

**Review by neo0351 — APPROVED** [review]

**Comment by serbinsh** [comment]
@neo0351 sorry making more changes I will let you know when ready.  Adding a fix for something @JulienLamour just found

File: R_Scripts/functions.R [hunk 1]
```diff
@@ -90,39 +90,52 @@ create_data_split <- function(approach=NULL, split_seed=123456789, prop=0.8,
 #--------------------------------------------------------------------------------------------------#
 
 #--------------------------------------------------------------------------------------------------#
-#' @param method Which approach to use to find optimal components, the PLS package of custom method
+#' @title pls_permutation
+#' @author Julien Lamour, Shawn P. Serbin
+#'
+pls_permutation <- function(dataset=NULL, maxComps=20, iterations=20, seg=100, prop=0.70) {
```
Comment (serbinsh) [inline R_Scripts/functions.R:96]: @JulienLamour @neo0351 new function to help generalize component selection

**Comment by serbinsh** [comment]
@neo0351 note updated/bug fixed component selection.  its pretty fast now because we are no longer running internal cross validation during the selection (duh) and instead using the fraction left off of when subseting the calibration data.  So now I think we can really speed up the NEON run./

**Comment by neo0351** [comment]
Great!!!  Are you going to merge this soon?  Or should I test it first? (is there a way to do that?)

**Comment by serbinsh** [comment]
@neo0351 merged. looking for testing now.  FYI - lots of changes in expanded_spectra-trait_reseco_lma_plsr_example.R that created outputs to save for the user that I/we will need to incorporate into the rest of the scripts.

**Comment by neo0351** [comment]
Let's test the one first, then we can edit the rest once we are happy.

**Comment by serbinsh** [comment]
Indeed. I also just realized we need to be sure we are saving coefficients properly in the new pls_permutation function.  specifically that we have the intercepts and that they are the first part of each column?  My brain is fried right now

**Pull request merged**